lib/spack/spack/bootstrap/_common.py

```diff
@@ -60,10 +61,18 @@ def _try_import_from_store(
             python, *_ = candidate_spec.dependencies("python-venv")
         else:
             python, *_ = candidate_spec.dependencies("python")
-        module_paths = [
-            os.path.join(candidate_spec.prefix, python.package.purelib),
-            os.path.join(candidate_spec.prefix, python.package.platlib),
-        ]
+
+        # if python is installed, ask it for the layout
+        if python.installed:
+            module_paths = [
+                os.path.join(candidate_spec.prefix, python.package.purelib),
+                os.path.join(candidate_spec.prefix, python.package.platlib),
+            ]
+        # otherwise search for the site-packages directory (assuming bootstapped from binaries)
```

Comment:
Can we make explicit that this case is for binary clingo with python-venv? For instance:
```suggestion
        # otherwise search for the site-packages directory
        # (clingo from binaries with truncated python-venv runtime)
```